airflow/models/mappedoperator.py

```diff
@@ -839,6 +841,8 @@ def unmap(self, resolve: None | Mapping[str, Any] | tuple[Context, Session]) ->
         from airflow.serialization.serialized_objects import SerializedBaseOperator
 
         op = SerializedBaseOperator(task_id=self.task_id, params=self.params, _airflow_from_mapped=True)
+        if "upstream_task_ids" not in self.operator_class:  # upstream_task_ids is not serialized
+            self.operator_class["upstream_task_ids"] = list(self.upstream_task_ids)
```

Comment:
This leads to the question… should we serialise it?

Reply:
Do you see any downsides, apart from slightly increasing the size of the serialised object and possibly, duplicating some information that can be deducted with downstream_task_ids?

Reply:
No; I think those are the reasons they are not serialised in the first place. (Plus also if you serialise both fields there’s always a risk they are not in sync.)

I think we should try to commit one way or the other though. If we want to serialise, we should always do that; if we don’t, we should get this information another way.

Reply:
I've tried quickly adding this field to serialisation, but it does not look trivial to me, I'm not so sure anymore that it's worth doing and I can't spend to much time on this right now unfortunately. If it can't be merged in current form let's close the PR for now - maybe I'll revisit in the future 😄 For now I'll advice the OL consumers to use AirflowDagFacet that has the taskTree with all the task dependencies in the DAG. Thanks for the comments and engagement !